sync/handler.go

```diff
@@ -64,29 +64,65 @@ func newMiniBlockRequestHandler(msh *mesh.Mesh, logger log.Log) func(msg []byte)
 			return nil
 		}
 
-		logger.Debug("return block %v", blk)
+		logger.Debug("return block %v", blk.ID())
 
 		return bbytes
 	}
 }
 
 func newTxsRequestHandler(msh *mesh.Mesh, logger log.Log) func(msg []byte) []byte {
 	return func(msg []byte) []byte {
-		var txid types.TransactionId
-		copy(txid[:], msg[:32])
-		logger.Info("handle tx request %v", hex.EncodeToString(txid[:]))
-		txs, _ := msh.GetTransactions([]types.TransactionId{txid})
+		var txids []types.TransactionId
+		err := types.BytesToInterface(msg, &txids)
+		if err != nil {
+			logger.Error("Error marshalling request", err)
+			return nil
+		}
+		logger.Info("handle tx request ")
+		txs, _ := msh.GetTransactions(txids)
+		if txs == nil {
+			logger.Error("Error handling transactions request message, with ids: %d", msg)
+			return nil
+		}
+
+		var transactions []types.SerializableTransaction
+		for _, value := range txs {
+			transactions = append(transactions, *value)
+		}
+
+		bbytes, err := types.InterfaceToBytes(transactions)
```

Comment:
serializing an array of pointers will work also, why do you need `var transactions []types.SerializableTransaction` ?

Reply:
tx's is a map so i had to convert

Reply:
this will cause a different order for the tx slice each time, is this expected behaviour?